src/components/shared/Map/MapPointer.tsx

```diff
@@ -4,35 +4,51 @@ import { Circle, Marker, Popup } from 'react-leaflet';
 
 import { type SensorData } from '@/types/models/SensorData';
 
+import { COLORS, LEVELS } from '../../../constants/index';
 import PointerPopup from './PointerPopup';
 
+type PollutionType = 'PM2.5' | 'PM10';
+
 interface MapPointerProps {
   center: [number, number];
   radius: number;
   text: string;
-  color: string;
   sensorData: SensorData;
 }
 
-function MapPointer({
-  center,
-  radius,
-  text,
-  color,
-  sensorData
-}: MapPointerProps) {
+function getCircleColor(
+  sensorData: SensorData,
+  pollutionName: PollutionType
+): string {
+  if (pollutionName != null) {
+    const pollutionValue = sensorData.data.find(
+      item => item.name === pollutionName
+    );
+    if (pollutionValue != null) {
+      const thresholds = (LEVELS as any)[pollutionName];
+      const idx = thresholds.findIndex(
+        (threshold: number) => pollutionValue.value < threshold
+      );
+      return (COLORS as any)[idx];
+    }
+  }
+  return (COLORS as any)[-1];
```

Comment:
JS (and TS) doesn't support negative integer indexing. You have to use `COLORS[COLORS.length - 1]` to get the last color (in this case it still won't work as `COLORS` is not an array but for arrays you can write as in the example).

Reply:
It was also highlighted in red when I did it "properly" the first time but ok.